github/actions/client.go

```diff
@@ -121,6 +138,37 @@ func NewClient(ctx context.Context, githubConfigURL string, creds *ActionsAuth,
 
 	retryClient.RetryMax = ac.retryMax
 	retryClient.RetryWaitMax = ac.retryWaitMax
+
+	if ac.rootCAs != nil {
+		transport, ok := retryClient.HTTPClient.Transport.(*http.Transport)
+		if !ok {
+			// this should always be true, because retryablehttp.NewClient() uses
+			// cleanhttp.DefaultPooledTransport()
+			return nil, fmt.Errorf("failed to get http transport from retryablehttp client")
+		}
+
+		if transport.TLSClientConfig == nil {
+			transport.TLSClientConfig = &tls.Config{}
+		}
+		transport.TLSClientConfig.RootCAs = ac.rootCAs
+		retryClient.HTTPClient.Transport = transport
+	}
+
+	if ac.tlsInsecureSkipVerify {
+		transport, ok := retryClient.HTTPClient.Transport.(*http.Transport)
+		if !ok {
+			// this should always be true, because retryablehttp.NewClient() uses
+			// cleanhttp.DefaultPooledTransport()
+			return nil, fmt.Errorf("failed to get http transport from retryablehttp client")
+		}
+
+		if transport.TLSClientConfig == nil {
+			transport.TLSClientConfig = &tls.Config{}
+		}
+		transport.TLSClientConfig.InsecureSkipVerify = true
+		retryClient.HTTPClient.Transport = transport
+	}
```

Comment:
```suggestion
	if ac.tlsInsecureSkipVerify {
		transport, ok := retryClient.HTTPClient.Transport.(*http.Transport)
		if !ok {
			// this should always be true, because retryablehttp.NewClient() uses
			// cleanhttp.DefaultPooledTransport()
			return nil, fmt.Errorf("failed to get http transport from retryablehttp client")
		}

		if transport.TLSClientConfig == nil {
			transport.TLSClientConfig = &tls.Config{}
		}
		transport.TLSClientConfig.InsecureSkipVerify = true
		retryClient.HTTPClient.Transport = transport
	} else if ac.rootCAs != nil {
		transport, ok := retryClient.HTTPClient.Transport.(*http.Transport)
		if !ok {
			// this should always be true, because retryablehttp.NewClient() uses
			// cleanhttp.DefaultPooledTransport()
			return nil, fmt.Errorf("failed to get http transport from retryablehttp client")
		}

		if transport.TLSClientConfig == nil {
			transport.TLSClientConfig = &tls.Config{}
		}
		transport.TLSClientConfig.RootCAs = ac.rootCAs
		retryClient.HTTPClient.Transport = transport
	}

```

Reply:
what about skip checking `ac.rootCAs` when `ac.tlsInsecureSkipVerify  == true`?

Reply:
I've just refactored that, do you mind taking a look? And also I don't think they should be exclusive, `InsecureSkipVerify` only kicks in if none of the provided CAs failed

Reply:
I like your refactor. 😄